analytics-webapps/src/main/webapp/vue-app/common-components/components/common/SettingDatePicker.vue

```diff
@@ -0,0 +1,175 @@
+<template>
+  <v-app id="analyticsDatePicker">
+    <v-menu
+      v-model="menu"
+      :close-on-content-click="false"
+      :nudge-right="40"
+      transition="scale-transition"
+      offset-y
+      min-width="auto">
+      <template v-slot:activator="{ on, attrs }">
+        <v-text-field
+          v-model="dateRangeText"
+          prepend-inner-icon="mdi-calendar"
+          class="pt-0 mt-0"
+          readonly
+          v-bind="attrs"
+          v-on="on" />
+      </template>
+      <v-date-picker
+        v-model="dates"
+        range
+        @input="selectedCustomDate()">
+        <div class="dateFooter footer">
+          <v-btn-toggle
+            class="d-flex flex-wrap justify-space-between"
+            tile
+            color="primary"
+            background-color="primary"
+            group>
+            <v-btn
+              v-for="(item, index) in periodOptions"
+              :key="index"
+              class="my-0"
+              :class="period && item.value === period.period && !useCustomDate? 'v-item--active v-btn--active' : ''"
+              small
+              @click="selectItem(item)">
+              {{ item.text }}
+            </v-btn>
+          </v-btn-toggle>
+        </div>
+      </v-date-picker>
+    </v-menu>
+  </v-app>
+</template>
+
+<script>
+export default {
+  props: ['value'],
+  data: () => ({
+    dates: [],
+    menu: false,
+    defaultValue: 'last6Months',
+    selectedItem: null,
+    useCustomDate:false
+  }),
+  computed:{
+    periodOptions() {
+      return [
+        {
+          value: 'lastYear',
+          text: this.$t('analytics.periodOptions.lastYear')
+        },
+        {
+          value: 'last6Months',
+          text: this.$t('analytics.periodOptions.last6Months'),
+        },
+        {
+          value: 'last3Months',
+          text: this.$t('analytics.periodOptions.last3Months')
+        },
+        {
+          value: 'lastMonth',
+          text: this.$t('analytics.periodOptions.lastMonth')
+        },
+        {
+          value: 'lastWeek',
+          text: this.$t('analytics.periodOptions.lastWeek')
+        },
+        {
+          value: 'last24h',
+          text: this.$t('analytics.periodOptions.last24h')
+        },
+      ];
+    },
+    period(){
+      return this.value;
+    },
+    dateRangeText () {
+      return this.dates.join('~');
+    },
+  },
+  watch: {
+    period(){
+      this.definePeriod();
+    },
+    menu(){
+      if (this.menu) {
+        this.$nextTick(() => {
+          $('.v-picker.v-card.v-picker--date.theme--light').addClass('cardDatePicker');
+          $('.v-btn.v-btn--active.v-btn--rounded.theme--light.accent').addClass('rageDatePicker');
+          $('.footer').addClass('footer');
+          if (this.useCustomDate){
+            $( '.dateFooter button' ).removeClass( 'v-item--active v-btn--active' );
+          }
```

Comment:
It's preferred to never using DOM manipulation inside Vue components
please review this and avoid adding jQuery DOM manipulations